### ISSUE TITLE
High CPU and Memory Utility

### ISSUE THREAD

**Issue by Daggerka** [xenolightning/AudioSwitcher_v1#2146]
My Set:
- [1.7.0.117 ] I have listed the Audio Switcher version
- [ Windows 10 Pro, 1607 ] I have listed my operating system, and version
- [Analog Speaker, HDMi TV, USB Headset, 2x Camera Mic, Analog Mic ] Include any details about the audio devices on the system

Audio Swither Utilize ~25% CPU Usage and lot of RAM memory. 

![audioswitherutility](https://cloud.githubusercontent.com/assets/407236/22445380/72d25954-e747-11e6-8a01-cfc1b3c087e4.JPG)

**Comment by xenolightning** [comment]
Does it happen immediately. Or is it after the application has been running for a long time?

**Comment by Wadera** [comment]
CPU o immediately. Memory utilization slowly increases (~1MB / 10 sec).

**Comment by xenolightning** [comment]
Ok, interesting. I wonder if Windows has changed something. I'll investigate it on the same build of Windows and get back to you.

In the meantime, can you screenshot the "Sound Dialog" from Control Panel, showing playback and recording devices, and showing any disabled/disconnected devices. High CPU might be caused by the number of devices on the system.

I may put together a new version tonight, and see if it helps with the problem.

Currently the website is down, which is getting fixed, but I'll upload any new versions to github first and link it in this issue.

**Comment by Wadera** [comment]
Btw, first service on top screen is sppsvc (Microsoft Software Protection Platform Service). 

It isn't first time so i decide to send ticket, but I can't reproduce it every time.

Screenshot below:

![1](https://cloud.githubusercontent.com/assets/407236/22446888/6d32f3fc-e74f-11e6-8903-652c264f61c7.JPG)
![2](https://cloud.githubusercontent.com/assets/407236/22446887/6d2fa198-e74f-11e6-9908-22f3d9e6f518.JPG)
![settings](https://cloud.githubusercontent.com/assets/407236/22446989/00a96c06-e750-11e6-8dc4-1be47a7db2d3.JPG)
![hotkeys](https://cloud.githubusercontent.com/assets/407236/22446991/00ac2590-e750-11e6-8700-2f9e63d815f8.JPG)
![about](https://cloud.githubusercontent.com/assets/407236/22446990/00aa3c9e-e750-11e6-8fc4-feb44ff06762.JPG)
![uptime](https://cloud.githubusercontent.com/assets/407236/22446886/6d2f33de-e74f-11e6-96d2-ab5b664f0596.JPG)

Thank you for your interest. Let me know if you need more information.

**Comment by xenolightning** [comment]
Awesome! That's a lot of info in those screenshots. 

I can't reproduce it immediately either. I wonder if it has to do with the volume monitors on application sound sessions.

A place to start looking might be if the system creates a large amount of sessions. You can view by opening the standard Windows volume mixer, it allows you to control volume per application.

This is one of the only places I could see the application using a large amount of CPU.

Just "brain dumping" here at the moment, not really anything concrete to start investigating.

**Comment by vlakoff** [comment]
Can you fix the bug by disconnecting some sound peripherals?
Just to know if the issue is related to a peripheral in particular.

**Comment by Stars-fury** [comment]
Hello, Xeno, firstly thank you for making such a useful app for me <3. Secondly, ever since I upgraded from win7 to win10 I have the same problem with CPU usage.
![task_manager](https://cloud.githubusercontent.com/assets/26323083/24454473/a8d9204e-1494-11e7-8145-1ea07bca8b71.png)

**Comment by sambul13** [comment]
It may be the same bug described[ here](https://github.com/xenolightning/AudioSwitcher_v1/issues/2173).

**Comment by xenolightning** [comment]
Yeah both are related. And it looks to be with the Audio Session code in the api.

Win10 seems to do something different, and takes longer to activate the audio sessions, or they are being activated incorrectly.

I'll be looking at fixing this, and releasing an updated version.

**Comment by panyann** [comment]
![Screenshot_1](https://user-images.githubusercontent.com/22815508/72172065-8a06b480-33d4-11ea-9365-b60ef1dd7dab.png)

I'de like to notice that the problem still exists. 100% usage of probably 1 CPU core on Windows 10 start...
